### PR TITLE
Sd 269 hp unpacker

### PR DESCRIPTION
**Pull request by weidenba**

### CODE REVIEW

File: src/plugins/unpacking/hp/internal/pjl_helper.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from pathlib import Path
+
+from common_helper_files import get_safe_name, write_binary_to_file
+
+NAME_FIELD_MAX = 102
+
+
+def get_pjl_commands(input_data):
+    pjl_instructions = []
+    tmp = list(re.finditer(b'@PJL [ =.!,?\w]+', input_data))
```
Comment (jstucke) [inline src/plugins/unpacking/hp/internal/pjl_helper.py]: `DeprecationWarning: invalid escape sequence \w`
should probably be a raw string `rb'...'`

File: src/plugins/unpacking/hp/internal/pjl_helper.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+def _get_end_postion_of_first_preamble(raw_binary):
+    tmp = re.search(b'\x25\x2d12345X\x0a?', raw_binary)
+    if tmp:
+        return tmp.end()
+    return 0
+
+
+def _get_name_of_upgrade(raw_binary, upgrade_command):
+    tmp = re.search(b'\xa8\x01([\w ]+)  [\w ]+', raw_binary[upgrade_command['end_offset']:upgrade_command['end_offset'] + NAME_FIELD_MAX])
```
Comment (jstucke) [inline src/plugins/unpacking/hp/internal/pjl_helper.py]: see above

File: src/plugins/unpacking/hp/internal/pjl_helper.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def _get_size_of_upgrade(upgrade_command):
+    tmp = re.match(b'SIZE=([0-9]+)', upgrade_command['value'])
+    tmp = int(tmp.group(1))
+    return tmp
+
+
+def _get_type_and_value(raw_command):
+    tmp = re.search(b'@PJL ([=\w]+) ?([ =.!,?\w]+)?', raw_command)
```
Comment (jstucke) [inline src/plugins/unpacking/hp/internal/pjl_helper.py]: see above

**Comment by weidenba** [comment]
tests run successfull: https://ci.die-weidenbachs.de/job/FACT_core_xenial-dev/522/console

**Pull request merged**